### PR TITLE
Fix DocService by handling Thrift typedefs as an UnresolvedClassInfo

### LINKED ISSUE #413: Failed to generate type infomation for typedef-ed type
When I define a type alias for exisiting thrift type like `typedef map<string, string> TypedefedMap`, it causes 
IllegalArgumentException during starting up thrift service.

Here is a test code to reproduce the error.
https://github.com/imasahiro/armeria/commit/4da9e8297b7e341cdfe3009afdef54eb29656728#diff-a2202e0d6b7d5b3fa867ca2bf3c2cf63R93

```
com.linecorp.armeria.server.thrift.ThriftServiceSpecificationGeneratorTest > servicesTest FAILED
    java.lang.IllegalArgumentException: unexpected field value type: 12
        at com.linecorp.armeria.server.thrift.ThriftServiceSpecificationGenerator.newTypeInfo(ThriftServiceSpecificationGenerator.java:346)
        at com.linecorp.armeria.server.thrift.ThriftServiceSpecificationGenerator.newFieldInfo(ThriftServiceSpecificationGenerator.java:291)
        at com.linecorp.armeria.server.thrift.ThriftServiceSpecificationGenerator.lambda$newFunctionInfo$9(ThriftServiceSpecificationGenerator.java:251)
        at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
        at java.util.Iterator.forEachRemaining(Iterator.java:116)
        at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
        at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
        at com.linecorp.armeria.server.thrift.ThriftServiceSpecificationGenerator.newFunctionInfo(ThriftServiceSpecificationGenerator.java:252)
        at com.linecorp.armeria.server.thrift.ThriftServiceSpecificationGenerator.newFunctionInfo(ThriftServiceSpecificationGenerator.java:225)
        at com.linecorp.armeria.server.thrift.ThriftServiceSpecificationGenerator.newServiceInfo(ThriftServiceSpecificationGenerator.java:158)
        at com.linecorp.armeria.server.thrift.ThriftServiceSpecificationGenerator.lambda$generate$4(ThriftServiceSpecificationGenerator.java:129)
        at java.util.ArrayList.forEach(ArrayList.java:1249)
        at com.linecorp.armeria.server.thrift.ThriftServiceSpecificationGenerator.generate(ThriftServiceSpecificationGenerator.java:126)
        at com.linecorp.armeria.server.thrift.ThriftServiceSpecificationGenerator.generate(ThriftServiceSpecificationGenerator.java:117)
        at com.linecorp.armeria.server.thrift.ThriftServiceSpecificationGeneratorTest.servicesTest(ThriftServiceSpecificationGeneratorTest.java:102)
```

### PR DESCRIPTION
**Pull request by trustin**
Motivation:

The FieldValueMetaData of a typedef'd field does not provide enough
information to infer its effective type. Translate typedef'd field into
a new type called UnresolvedClassInfo so that
ThriftServiceSpecificationGenerator does not fail at least.

Modifications:

- Add UnresolvedClassInfo
- Modify ThriftServiceSpecificationGenerator so that it does not fail on
  a typedef'd field
- Revise @imasahiro's test case to reflect the modification of
  ThriftServiceSpecificationGenerator

Result:

- DocService does not fail to start when typedef is used in .thrift
  files.
- Fixes #413

### CODE REVIEW

File: core/src/main/java/com/linecorp/armeria/server/docs/UnresolvedClassInfo.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2015 LINE Corporation
```
Comment (imasahiro) [inline core/src/main/java/com/linecorp/armeria/server/docs/UnresolvedClassInfo.java]: 2015 -> 2017

File: core/src/main/java/com/linecorp/armeria/server/docs/UnresolvedClassInfo.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import javax.annotation.Nullable;
+
+import com.google.common.collect.ImmutableList;
+
+/**
+ * Metadata about a struct type.
```
Comment (anuraaga) [inline core/src/main/java/com/linecorp/armeria/server/docs/UnresolvedClassInfo.java]: Should update this javadoc

File: thrift/src/test/thrift/main.thrift [hunk 3]
```diff
@@ -77,14 +77,26 @@ struct FooStruct {
     12: map<string, FooEnum> mapVal,
     13: set<FooUnion> setVal,
     14: list<string> listVal,
+    15: optional FooStruct selfRef
 }
 
+typedef FooStruct           TypedefedStruct
```
Comment (anuraaga) [inline thrift/src/test/thrift/main.thrift]: How about some plain typedefs too?

```java
typedef i32 Size
```

Comment (trustin) [inline thrift/src/test/thrift/main.thrift]: It seems like Thrift compiler translates them for us simply.

Comment (trustin) [inline thrift/src/test/thrift/main.thrift]: Added `typedef string TypedefedString` instead.

**Pull request merged**